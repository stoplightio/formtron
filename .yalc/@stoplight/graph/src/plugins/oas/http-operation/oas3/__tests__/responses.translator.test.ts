jest.mock('../helpers/content.translator');
import { mockPassthroughImplementation } from '@stoplight/test-utils';
import { translateHeaderObject, translateMediaTypeObject } from '../helpers/content.translator';
import { translateToOas3Responses } from '../helpers/responses.translator';

describe('translateToOas3Responses', () => {
  beforeEach(() => {
    mockPassthroughImplementation(translateHeaderObject, translateMediaTypeObject);
  });

  test('given empty dictionary should return empty array', () => {
    expect(translateToOas3Responses({})).toMatchSnapshot();
  });

  test('given a response in dictionary should translate', () => {
    expect(
      translateToOas3Responses({
        default: {
          content: {
            'fake-content-type': {},
          },
          description: 'descr',
          headers: {
            'fake-header-name-1': {},
            'fake-header-name-2': {},
          },
        },
        200: {
          content: {
            'fake-content-type-200': {},
          },
          description: 'descr 200',
          headers: {
            'fake-header-name-200': {},
          },
        },
      })
    ).toMatchSnapshot();
  });
});
