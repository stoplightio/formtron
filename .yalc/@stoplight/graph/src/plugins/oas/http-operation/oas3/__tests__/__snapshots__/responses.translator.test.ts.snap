// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translateToOas3Responses given a response in dictionary should translate 1`] = `
Array [
  Object {
    "code": "200",
    "contents": Array [
      "translateMediaTypeObject({}, fake-content-type-200)",
    ],
    "description": "descr 200",
    "headers": Array [
      "translateHeaderObject({}, fake-header-name-200)",
    ],
  },
  Object {
    "code": "default",
    "contents": Array [
      "translateMediaTypeObject({}, fake-content-type)",
    ],
    "description": "descr",
    "headers": Array [
      "translateHeaderObject({}, fake-header-name-1)",
      "translateHeaderObject({}, fake-header-name-2)",
    ],
  },
]
`;

exports[`translateToOas3Responses given empty dictionary should return empty array 1`] = `Array []`;
